tests/v3/index.test.ts

```diff
@@ -0,0 +1,125 @@
+import prettier from 'prettier';
+
+import { OpenAPI3 } from '../../src/types';
+import v3, { WARNING_MESSAGE, PRETTIER_OPTIONS } from '../../src/v3';
+
+// test helper: don’t throw the test due to whitespace differences
+function format(types: string): string {
+  return prettier.format([WARNING_MESSAGE, types.trim()].join('\n'), PRETTIER_OPTIONS);
+}
+
+describe('types', () => {
+  it('string', () => {
+    const schema: OpenAPI3 = {
+      openapi: '3.0',
+      components: {
+        schemas: {
+          object: { properties: { string: { type: 'string' } }, type: 'object' },
+          string: { type: 'string' },
+          string_ref: { $ref: '#/components/schemas/string' },
+        },
+      },
+    };
+    expect(v3(schema)).toBe(
+      format(`
+      export interface components {
+        schemas: {
+          object: { string: string };
+          string: string;
+          string_ref: components['schemas']['string'];
+        }
+      }`)
+    );
+  });
+
+  it('number', () => {
+    const schema: OpenAPI3 = {
+      openapi: '3.0',
+      components: {
+        schemas: {
+          object: {
+            properties: { integer: { type: 'integer' }, number: { type: 'number' } },
+            type: 'object',
+          },
+          number: { type: 'number' },
+          number_ref: { $ref: '#/components/schemas/number' },
+        },
+      },
+    };
+    expect(v3(schema)).toBe(
+      format(`
+      export interface components {
+        schemas: {
+          object: { integer: number; number: number }
+          number: number;
+          number_ref: components['schemas']['number'];
+        }
+      }`)
+    );
+  });
```

Comment:
👆 Same for the `number` and `boolean` tests too 👇  (note that every test kinda has multiple parts to each—it’s OK to focus on one line at a time for each test; you won’t get all in one pass!)